includes/class-wc-payments-woopay-direct-checkout.php

```diff
@@ -71,6 +74,36 @@ public function scripts() {
 			return;
 		}
 
+		// Enqueue the WCPay common config script only if it hasn't been enqueued yet.
+		// This may happen when Direct Checkout is being enqueued on pages that are not the cart page,
+		// such as the home and shop pages.
+		if ( did_filter( 'wcpay_payment_fields_js_config' ) === 0 ) {
```

Comment:
If this filter ([ref](https://github.com/Automattic/woocommerce-payments/blob/634c04206194f9d38e24dce944d49d6f4f65b013/includes/class-wc-payments-checkout.php#L290)) isn't applied, `wcpayConfig` won't be registered in the frontend and will be evaluated to `undefined`. Therefore, we need to enqueue it so that the direct checkout flow doesn't break.